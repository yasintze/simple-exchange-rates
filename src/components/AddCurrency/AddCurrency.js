// @flow
import React from "react";
import Button from "@material-ui/core/Button";
import FormControl from "@material-ui/core/FormControl";
import InputLabel from "@material-ui/core/InputLabel";
import Select from "@material-ui/core/Select";
import MenuItem from "@material-ui/core/MenuItem";
import Paper from "@material-ui/core/Paper";
import { withStyles } from "@material-ui/core/styles";

type Props = {
  data: Array<Object>,
  selected: string,
  onChange: Function,
  onClick: Function,
  classes: Object
};

const styles = theme => ({
  button: {
    margin: theme.spacing.unit,
    maxWidth: 240
  },
  card: {
    minWidth: 275
  },
  formControl: {
    margin: theme.spacing.unit,
    width: "100%"
  },
  paper: {
    padding: theme.spacing.unit * 2,
    textAlign: "center",
    color: theme.palette.text.secondary,
    backgroundColor: "transparent",
    minHeight: "calc(100vh - 180px)",
    maxWidth: 768,
    margin: "auto"
  },
  textBlack: {
    color: "black"
  }
});

const AddCurrency = (props: Props) => {
  const { data, selected, onChange, onClick, classes } = props;
  return (
    <Paper className={classes.paper}>
      <FormControl className={classes.formControl}>
        <InputLabel htmlFor="add-currency">Select currency</InputLabel>
        <Select
          value={selected}
          onChange={onChange}
          inputProps={{
            name: "add-currency",
            id: "add-currency"
          }}
        >
          <MenuItem value="">
            <em>Add more currency</em>
          </MenuItem>
          {Object.keys(data).map(item => (
            <MenuItem key={item} value={item}>
              {item}
            </MenuItem>
          ))}
        </Select>
      </FormControl>
      <Button
        variant="contained"
        color="primary"
        className={classes.button}
        onClick={onClick}
      >
        {"Add Currency"}
      </Button>
    </Paper>
  );
};

export default withStyles(styles)(AddCurrency);
