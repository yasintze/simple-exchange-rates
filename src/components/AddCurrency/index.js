import AddCurrency from "./AddCurrency";

export default AddCurrency;
