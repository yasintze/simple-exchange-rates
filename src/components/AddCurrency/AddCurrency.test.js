// @flow
import React from "react";
import { shallow } from "enzyme";

import AddCurrency from "./AddCurrency";

describe("AddCurrency component", () => {
  it("render correctly", () => {
    const wrapper = shallow(<AddCurrency />);
    expect(wrapper).toMatchSnapshot();
  });
});
