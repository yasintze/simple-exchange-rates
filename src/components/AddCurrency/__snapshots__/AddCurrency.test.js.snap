// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCurrency component render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(AddCurrency) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "classes": Object {
        "button": "AddCurrency-button-1",
        "card": "AddCurrency-card-2",
        "formControl": "AddCurrency-formControl-3",
        "paper": "AddCurrency-paper-4",
        "textBlack": "AddCurrency-textBlack-5",
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "classes": Object {
          "button": "AddCurrency-button-1",
          "card": "AddCurrency-card-2",
          "formControl": "AddCurrency-formControl-3",
          "paper": "AddCurrency-paper-4",
          "textBlack": "AddCurrency-textBlack-5",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
